app/src/main/java/pl/gittobefit/WorkoutDisplay/dialog/EditExerciseDialog.java

```diff
@@ -0,0 +1,141 @@
+        this.position = position;
+        this.exercisesExecutionArrayList = exercisesExecutionArrayList;
+        this.trainingID = trainingID;
+        this.exerciseName = exerciseName;
+        this.exerciseExecutionPOJODBS = exerciseExecutionPOJODBS;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        LayoutInflater inflater = getActivity().getLayoutInflater();
+        View view = inflater.inflate(R.layout.layout_edit_exercise_dialog, null);
+
+        seriesNumberPicker = view.findViewById(R.id.seriesNumberPicker);
+        seriesNumberPicker.setMaxValue(10);
+        seriesNumberPicker.setValue(exercisesExecutionArrayList.get(position).getSeries());
+        seriesNumberPicker.setMinValue(1);
+        seriesNumberPicker.setWrapSelectorWheel(false);
+        seriesNumberPicker.setOnValueChangedListener(this);
+
+        countNumberPicker = view.findViewById(R.id.countNumberPicker);
+
+        if (exercisesExecutionArrayList.get(position).getTime() > 20) {
+            countNumberPicker.setMaxValue(90);
+            countNumberPicker.setMinValue(10);
+            countNumberPicker.setValue(exercisesExecutionArrayList.get(position).getTime());
+        } else {
+            countNumberPicker.setMaxValue(50);
+            countNumberPicker.setMinValue(1);
+            countNumberPicker.setValue(exercisesExecutionArrayList.get(position).getCount());
+        }
+
+
+        countNumberPicker.setWrapSelectorWheel(false);
+        countNumberPicker.setOnValueChangedListener(this);
+
+        TextView seriesCount = view.findViewById(R.id.seriesCount);
+        if (scheduleType.equals("SERIES"))
+        {
+            seriesCount.setText("Ilość serii");
+        }
+        else if (scheduleType.equals("CIRCUIT"))
+        {
+            seriesCount.setText("Ilość obwodów");
+        }
+
+        TextView count_time = view.findViewById(R.id.count_time);
+        if (exercisesExecutionArrayList.get(position).getTime() > 20)
+        {
+            count_time.setText("Czas trwania");
+        }
+        else {
+            count_time.setText("Ilość powtórzeń");
+        }
+        InitiationTrainingDisplayLayoutViewModel model = new ViewModelProvider(requireActivity()).get(InitiationTrainingDisplayLayoutViewModel.class);
+
+        builder.setView(view)
+                .setTitle(exerciseName)
+                .setNegativeButton(getString(R.string.admit_changes), (dialog, which) ->
+                {
+                    if (scheduleType.equals("CIRCUIT"))
+                    {
+                        for (ExerciseExecutionPOJODB item: exercisesExecutionArrayList
+                             ) {
+                            item.setSeries(seriesNumberPicker.getValue());
+                            model.getCurrentSeries().setValue(seriesNumberPicker.getValue());
```

Comment:
dlaczego robisz to w modelu i robisz to tutaj wydaje sie ze najlatwiej byloby to robic w modelu ktory by automatycznie tez przez repozytorium zapisywal to w bazie